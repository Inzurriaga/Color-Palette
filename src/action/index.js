export const toggleProjectDisplay  = () => ({
    type: "TOGGLE_PROJECT"
})

export const generateColorPalette = (palette) => ({
    type: "GENERATED_PALETTE",
    palette
})

export const toggleSaveModal = () => ({
    type: "TOGGLE_SAVE"
})

export const initialProjects = (projects) => ({
    type: "INITIAL_PROJECTS",
    projects
})

export const addNewProject = (project) => ({
    type: "ADD_PROJECT",
    project
})

export const initialPalette = (palettes) =>({
    type: "INITIAL_PALETTE",
    palettes
})

export const addNewPalette = (palette) => ({
    type: "ADD_PALETTE", 
    palette
})

export const deleteProject = (id) => ({
    type: "DELETE_PROJECT",
    id
})

export const deletePalette = (id) => ({
    type: "DELETE_Palette",
    id
})

export const deleteProjectPalettes = (projectId) => ({
    type: "DELETE_PROJECT_PALETTE",
    projectId
})