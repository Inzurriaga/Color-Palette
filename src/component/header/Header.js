import React, { Component } from "react";
import { connect } from "react-redux";
import { toggleProjectDisplay, generateColorPalette } from "../../action"

export  class Header extends Component{
    constructor(){
        super();
        this.state = {
            test: "hello im a test"
        }
    }

    render(){
        return(
            <header>
                <div>
                    <h1>Palette Picker</h1>
                </div>
                <nav>
                    <button className="refresh" onClick={this.props.generateColorPalette}></button>
                    <button className="save"></button>
                    <button className="projects" onClick={this.props.toggleProjectDisplay}></button>
                </nav>
            </header>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    toggleProjectDisplay: () => dispatch(toggleProjectDisplay()),
    generateColorPalette: () => dispatch(generateColorPalette())
})

export default connect(null, mapDispatchToProps)(Header)