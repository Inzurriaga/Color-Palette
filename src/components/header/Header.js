import React, { Component } from "react";
import { connect } from "react-redux";
import { toggleProjectDisplay, generateColorPalette, toggleSaveModal } from "../../action"

export  class Header extends Component{

    render(){
        const colorPalette = this.props.colorPalette
        return(
            <header>
                <div>
                    <h1>Palette Picker</h1>
                </div>
                <nav>
                    <button className="refresh" onClick={() => {this.props.generateColorPalette(colorPalette)}}></button>
                    <button className="save" onClick={this.props.toggleSaveModal}></button>
                    <button className="projects" onClick={this.props.toggleProjectDisplay}></button>
                </nav>
            </header>
        )
    }
}

export const mapStateToProps = (state) => ({
    colorPalette: state.colorPalette
})

export const mapDispatchToProps = (dispatch) => ({
    toggleProjectDisplay: () => dispatch(toggleProjectDisplay()),
    generateColorPalette: palette => dispatch(generateColorPalette(palette)),
    toggleSaveModal: () => dispatch(toggleSaveModal())
})


export default connect(mapStateToProps, mapDispatchToProps)(Header)