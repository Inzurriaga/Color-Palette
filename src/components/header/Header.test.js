import React from "react"
import { shallow } from "enzyme"
import { toggleProjectDisplay, generateColorPalette, toggleSaveModal } from "../../action"
import { Header, mapDispatchToProps, mapStateToProps } from "./Header"

const mockToggleProjectDisplay = jest.fn()
const mockGenerateColorPalette = jest.fn()
const mockToggleSaveModal = jest.fn()
const mockColorPalette = [{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
}]

describe("App", () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <Header colorPalette={mockColorPalette}
                    toggleProjectDisplay={mockToggleProjectDisplay}
                    generateColorPalette={mockGenerateColorPalette}
                    toggleSaveModal={mockToggleSaveModal}/>
        )
    })

    describe("App", () => {

        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot();
        })

        it("should call generateColorPalette when button is click", () => {
            wrapper.find(".refresh").simulate("click");
            expect(mockGenerateColorPalette).toHaveBeenCalled();
        })

        it("should call toggleSaveModal when button is click", () => {
            wrapper.find(".save").simulate("click");
            expect(mockToggleSaveModal).toHaveBeenCalled();
        })

        it("should call toggleProjectDisplay when button is click", () => {
            wrapper.find(".projects").simulate("click");
            expect(mockToggleProjectDisplay).toHaveBeenCalled()
        })

    })
    describe("mapDispatchToProps", () => {

        it('calls dispatch with an toggleProjectDisplay action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = toggleProjectDisplay()
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.toggleProjectDisplay()
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

          it('calls dispatch with an toggleSaveModal action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = toggleSaveModal()
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.toggleSaveModal()
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

          it('calls dispatch with an generateColorPalette action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = generateColorPalette(mockColorPalette)
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.generateColorPalette(mockColorPalette)
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

    })
    describe("mapStateToProps", () => {

        it("should only return waht is needed from the store", () => {
            const mockStore = {
                colorPalette: [{},{},{}],
                project: [{},{},{}]
            }
            const expected = {
                colorPalette: [{},{},{}]
            }
            const result = mapStateToProps(mockStore)
            expect(result).toEqual(expected)
        })
        
    })
})