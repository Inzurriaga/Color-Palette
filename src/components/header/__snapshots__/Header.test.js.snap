// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<header>
  <div>
    <h1>
      Palette Picker
    </h1>
  </div>
  <nav>
    <button
      className="refresh"
      onClick={[Function]}
    />
    <button
      className="save"
      onClick={[MockFunction]}
    />
    <button
      className="projects"
      onClick={[MockFunction]}
    />
  </nav>
</header>
`;
